python/ray/experimental/client/dataclient.py

```diff
@@ -0,0 +1,101 @@
+"""
+This file implements a threaded stream controller to abstract a data stream
+back to the ray clientserver.
+"""
+import logging
+import queue
+import threading
+
+from typing import Any
+from typing import Dict
+
+import ray.core.generated.ray_client_pb2 as ray_client_pb2
+import ray.core.generated.ray_client_pb2_grpc as ray_client_pb2_grpc
+
+logger = logging.getLogger(__name__)
+
+
+class DataClient:
+    def __init__(self, channel, client_id):
+        """Initializes a thread-safe datapath over a Ray Client gRPC channel.
+
+        Args:
+            channel: connected gRPC channel
+        """
+        self.channel = channel
+        self.request_queue = queue.Queue()
+        self.data_thread = self._start_datathread()
+        self.ready_data: Dict[int, Any] = {}
+        self.cv = threading.Condition()
+        self._req_id = 0
+        self._max_id = 100000
```

Comment:
Why not let it increment forever? It's unlikely we hit 2 billion requests on a client.

Reply:
My philosophy here was that we're unlikely to have 100,000 requests simultaneously in flight. But I'll make it a real constant in the file, so we can have billions in-flight.

The reason not to let it increment forever is because someone out there is going to test that limit. I'm coding defensively. Sure, once-a-millisecond seems excessive, but INT32_MAX milliseconds is just shy of a month and I've seen crazier things.

Alternatively, I can make the field an int64 and call it a day, but, again, it's there to disambiguate in-flight requests -- so why not use it like that?